update definition of CVSS score (#25892)
DataDog/documentation: content/en/security/application_security/software_composition_analysis/_index.md

The Datadog SCA Library Inventory helps you understand the list of libraries and its versions that compose your application. To access the Library Explorer, navigate to [**Security** > **Application Security** > **Catalog** > **Libraries**][8].

> Since Datadog SCA covers your software development lifecycle from code to production, it detects libraries throughout the lifecycle of an application and flags vulnerabilities, risks, licenses, and more.

Reviewer: ```suggestion
With Datadog SCA spanning your software development lifecycle from code to production, it detects libraries throughout the lifecycle of an application and alerts you to vulnerabilities, risks, licenses, and more.
```

- Runtime analysis in your deployed services (runtime point of view)

Combining both techniques monitors open source libraries end-to-end, from the code repository commit (static point of view), to the applications running in production (runtime point of view).

To switch to the code repository commit point of view, select **Static**. The static view shows vulnerabilities from the _source code_ in your repositories.

To switch to the _real-time_ point of view for the applications already running, select **Runtime**. The runtime view is the live view of the services monitored by Datadog.

{{< img src="/security/application_security/software_composition_analysis/asm_sca_vulnerabilities.png" alt="Software Composition Analysis (SCA) explorer page showing vulnerabilities sorted by static or runtime." style="width:100%;" >}}

Select a specific vulnerability to see its details, including the affected services, severity breakdown score, and recommended remediation steps.

On the Details Explorer for a vulnerability, you can view impacted infrastructure. This view gives you better insights to your overall attack exposure.

## Datadog Severity Score

> Each vulnerability has a defined base severity score. To help prioritize remediation, Datadog adjusts the base CVSS score into the Datadog Severity Score using
> evidence of suspicious requests or attacks, the business sensitivity or exposure of the environment to the Internet, and the risk of a potential successful exploit.

Reviewer: ```suggestion
Each vulnerability has a defined base severity score. To assist in prioritizing remediation, Datadog modifies the base CVSS score into the Datadog Severity Score by considering evidence of suspicious requests or attacks, the business sensitivity or internet exposure of the environment, and the risk of a successful exploit.
```

The Datadog SCA Library Inventory helps you understand the list of libraries and its versions that compose your application. To access the Library Explorer, navigate to [**Security** > **Application Security** > **Catalog** > **Libraries**][8].

> Since Datadog SCA covers your software development lifecycle end-to-end, the libraries are detected throughout the entire lifecycle of the application. The library inventory contains everything you need to know about the libraries, including name and version, and other risk aspects such as licenses and quality aspects.

Reviewer 2: "Since Datadog SCA covers your software development lifecycle from code to production, it detects libraries throughout the lifecycle of an application and flags vulnerabilities, risks, licenses, and more."

- Runtime analysis in your deployed services (runtime point of view)

Combining both techniques monitors open source libraries end-to-end, from the code repository commit (static point of view), to the applications running in production (runtime point of view).

To switch to the code repository commit point of view, select **Static**. The static view shows vulnerabilities from the _source code_ in your repositories.

To switch to the _real-time_ point of view for the applications already running, select **Runtime**. The runtime view is the live view of the services monitored by Datadog.

{{< img src="/security/application_security/software_composition_analysis/asm_sca_vulnerabilities.png" alt="Software Composition Analysis (SCA) explorer page showing vulnerabilities sorted by static or runtime." style="width:100%;" >}}

Select a specific vulnerability to see its details, including the affected services, severity breakdown score, and recommended remediation steps.

On the Details Explorer for a vulnerability, you can view impacted infrastructure. This view gives you better insights to your overall attack exposure.

## Datadog Severity Score

> Each vulnerability has a defined base severity score. To help prioritize remediation, Datadog provides an enhanced severity score (known as "Datadog Severity Score") using existing attacks

Reviewer 2: "Each vulnerability has a defined base severity score from [CVSS](https://www.first.org/cvss/)."

Author: the score is not from CVSS. CVSS is the method

Reviewer 3: I agree with Kassen, we should clearly state that the base score is following CVSS.

Also, I often heard customers didn't understand the Datadog score is following CVSS as well, we should state it clearly.

For example:
`Datadog adjust the base CVSS vector by modifying the temporal and environmental score.`

We could demystify further by explaining how the base CVSS vector is adjusted, modifier by modifier.

For example:
Considering [this CVSS](https://www.first.org/cvss/calculator/3.0#CVSS:3.0/AV:N/AC:H/PR:N/UI:R/S:C/C:L/I:L/A:L/E:H/RL:U/RC:C/CR:H/IR:H/AR:H/MAV:N/MAC:H/MPR:N/MUI:R) having
 * a vector `AV:N/AC:H/PR:N/UI:R/S:C/C:L/I:L/A:L/E:H/RL:U/RC:C/CR:H/IR:H/AR:H/MAV:N/MAC:H/MPR:N/MUI:R`
 * a score of 7.2
 * a `High` severity

If the vulnerability is not detected on a **production environment**, then the environmental score is updated as follow:
 - The Confidentiality Requirement (CR) is set to `Low`
 - The Integrity Requirement (IR) is set to `Low`
 - The Availability Requirement (AR) is set to `Low`

The Datadog score results in:
 * [this CVSS](https://www.first.org/cvss/calculator/3.0#CVSS:3.0/AV:N/AC:H/PR:N/UI:R/S:C/C:L/I:L/A:L/E:H/RL:U/RC:C/CR:L/IR:L/AR:L/MAV:N/MAC:H/MPR:N/MUI:R)
 * having this vector `AV:N/AC:H/PR:N/UI:R/S:C/C:L/I:L/A:L/E:H/RL:U/RC:C/CR:L/IR:L/AR:L/MAV:N/MAC:H/MPR:N/MUI:R`
 * a score of 4.0
 * a `Medium` severity

@kassenq @juli1 what do you think?

If we want to follow this path, @juli1 you can see how the CVSS is adjusted in [this class](https://github.com/DataDog/logs-backend/blob/a0c521c50875a7863c4a4648c98d1fdf982ba437/domains/cloud-security-platform/application-security/libs/vulnerability-common/src/main/java/com/dd/appsec/vulnerability/cvss/CvssModifier.java#L122)

Reviewer 4: CVSS defines how to change the vuln attributes in score (https://www.first.org/cvss/v3.1/specification-document#1-2-Scoring) and how to map this score to severity (https://www.first.org/cvss/v3.1/specification-document#Qualitative-Severity-Rating-Scale)

Reviewer 4: I would say that the Datadog score also leverages CVSS

and the business sensitivity of the environment where the vulnerability is detected.

For example, if no production environment is detected, the severity is reduced.

The adjusted vulnerability score includes the full context of each service:

- The original vulnerability severity
- Evidence of suspicious requests or attacks
- Sensitive or internet-exposed environments
- Risk of a potential successful exploit

> Datadog shows how the base CVSS score is updated into a Datadog Severity Score based on your execution environment.

Reviewer 3: not only the execution environment but also the exploitability of the CVE:
 * is there proof this CVE has been exploited ? e.g. from https://www.exploit-db.com/
 * is there a high probability this CVE would be exploited ? from [EPSS](https://www.first.org/epss/)
